### PR TITLE
.Net Core port

### PR DESCRIPTION
**Pull request by virzak**
Porting this project to .Net Core

### CODE REVIEW

**Comment by emgucv** [comment]
@virzak, Thanks for contributing. 

However, some of the changes has already been committed to the Git repo earlier:
b64c966ebe82fe15dc07101a53a88f0c349c444a
52e332905243770e73e5ef4964bcdf35a635d735

.Net standard profile is also compatible with .net core. So implementing .net standard will make it possible support more profiles. 

You can use the .net standard dll in .net core projects.

We are also working on adding the .netstandard profile to the standard nuget release. It will be slightly different from what you have implemented. We started experimenting with the Emgu TF project and it will be implemented on Emgu CV as well.

Sorry for the delay in porting to .netstandard (also compatible with .netcore). We wants to do it right so there is a considerable amount of planning.

**Comment by virzak** [comment]
@emgucv Totally understand.

Would have waited longer, however our project was held up by this library. Needed to at least release something in the meantime (for our impatient needs) to move forward.

Would love to participate in any discussions to make this change right for the future. Go you have a gitter for this project?

File: Emgu.CV/Videoio/ICapture.cs [hunk 1]
```diff
@@ -1,7 +1,7 @@
 //----------------------------------------------------------------------------
 //  Copyright (C) 2004-2019 by EMGU Corporation. All rights reserved.       
 //----------------------------------------------------------------------------
-#if !(__ANDROID__ || __UNIFIED__ || NETFX_CORE || UNITY_WSA || NETSTANDARD1_4 || UNITY_ANDROID || UNITY_IOS || UNITY_EDITOR || UNITY_STANDALONE || NETCOREAPP1_1)
+#if !(__ANDROID__ || __UNIFIED__ || NETFX_CORE || UNITY_WSA || NETSTANDARD1_4 || UNITY_ANDROID || UNITY_IOS || UNITY_EDITOR || UNITY_STANDALONE || NETCOREAPP1_1 || NETCOREAPP)
```
Comment (mconnew) [inline Emgu.CV/Videoio/ICapture.cs]: The items affected by this are available for .NET Standard/.NET Core and are needed to use a WCF Client. The types in the code base that need usage removed are mentioned in [this](https://github.com/emgucv/emgucv/issues/1#issuecomment-525006712) comment. I believe all the attributes this change removes are in the nuget package System.ServiceModel.Primitives.

Comment (virzak) [reply Emgu.CV/Videoio/ICapture.cs]: Thanks Matt,

Added the reference to .Net Core 3.0.
Looks like XmlSerializerFormatAttribute is [not available](https://apisof.net/catalog/System.ServiceModel.XmlSerializerFormatAttribute) before .Net Standard 1.6. Not sure if the maintainers want to target it as well.

Comment (mconnew) [reply Emgu.CV/Videoio/ICapture.cs]: The oldest .NET Core version supported is .NET Core 2.1 so all currently supported version of .NET Core support at least .NET Standard 2.0

**Comment by emgucv** [comment]
Any reason not to use the .Netstandard dll in .Net Core? 
Our 4.1.1 release comes with a Emgu.CV.World.Netstandard1_4.dll that use .net standard profile and can be used in .Net Core 1.0 and higher.

**Comment by virzak** [comment]
@emgucv So you mean not multitargeting? 

Are there any features in Emgu.CV.World which are platform specific which would be lost?
Didn't seem to me that ZedGraph is really being used in this assembly.
For service model and [XmlSerializerFormatAttribute](https://apisof.net/catalog/System.ServiceModel.XmlSerializerFormatAttribute) it should be available in .Net Standard 1.6, so perhaps it is worth upgrading or multitarget to two .net standard versions?

Targeting .Net Standard only will reduce the complexity, so not a bad idea, IMHO.

Also consider @terrajobst https://github.com/emgucv/emgucv/issues/1#issuecomment-551304222 .

**Comment by emgucv** [comment]
We are working on an upgrade to .Netstandard 2.0 in the next few days. It is a big change that will have conflicts with the current pull request. But we think supporting a single ".Netstandard 2.0" profile will be easier to maintain than multi-targeting. It will be compatible with .net Core 2.0 and higher according to the following table:
https://docs.microsoft.com/en-us/dotnet/standard/net-standard

We are targeting a big commit later this week or earlier next week. We shall be able to make the new ".Netstandard 2.0" profile support Windows, Linux, MacOS and Android. iOS will not be supported in the ".Netstandard 2.0" profile due to the different PInvoke entry point ("__Internal" entry point) in this up coming commit.

**Comment by d8ahazard** [comment]
Hi @virzak - I'm trying to utilize your port and having some issues, do you think we could chat somewhere regarding setup?

**Comment by emgucv** [comment]
The 4.2.0 release had converted Emgu.CV nuget package to target .Net Standard 2.0. 

Closing this merge request now.

**Pull request closed without merging**